aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -21,7 +21,7 @@ static inline Device ensure_has_index(Device device) {
 
 static inline Tensor to_impl(const Tensor& self, const TensorOptions& options, bool non_blocking, bool copy, c10::optional<c10::MemoryFormat> optional_memory_format) {
   auto memory_format =
-      optional_memory_format.value_or(MemoryFormat::Contiguous);
+      optional_memory_format.value_or(MemoryFormat::Preserve);
```

Comment:
wasn't this broken before?  Because if I didn't pass in a memory format, memory_format would be set to contiguous.

But then suggest_memory_format wouldn't be contiguous, so I create a new tensor and copy.  But that doesn't seem like the correct semantics for the function.

Reply:
Indeed, looks like (in master) if we have channels last tensor and call .to to same dtype etc. we will get contiguous output instead of self. 

This PR will solve the issue.

Reply:
is that behavior in 1.4?  Should we introduce a patch to fix it?

Reply:
it would probably be good to fix that in a separate patch -- in case this gets reverted (happens with BC changes), we can have the bug fix in without this.

Reply:
and seems worthy of a test :).